src/components/containers/TraceAccordion.js

```diff
@@ -10,78 +10,155 @@ import {Tab, Tabs, TabList, TabPanel} from 'react-tabs';
+    if (this.props.useFullData) {
+      return filteredTraces.map((d, i) => {
+        return (
+          <TraceFold
+            key={i}
+            traceIndexes={[d.index]}
+            canDelete={this.props.canAdd}
+            messageIfEmpty={this.props.messageIfEmptyFold}
+            fullDataArrayPosition={[filteredTracesFullDataPositions[i]]}
+          >
+            {this.props.children}
+          </TraceFold>
+        );
+      });
     }
-    const tracesByGroup = filteredData.reduce((allTraces, nextTrace, index) => {
-      const traceType = plotlyTraceToCustomTrace(nextTrace);
-      if (!allTraces[traceType]) {
-        allTraces[traceType] = [];
-      }
-      allTraces[traceType].push(index);
-      return allTraces;
-    }, {});
 
-    const groupedTraces = Object.keys(tracesByGroup).map((traceType, index) => {
+    return filteredTraces.map((d, i) => {
       return (
         <TraceFold
-          key={index}
-          traceIndexes={tracesByGroup[traceType]}
-          name={traceType}
+          key={i}
+          traceIndexes={[i]}
+          canDelete={this.props.canAdd}
+          messageIfEmpty={this.props.messageIfEmptyFold}
         >
           {this.props.children}
         </TraceFold>
       );
     });
+  }
+
+  renderCanAddPanel(filteredTraces, filteredTracesFullDataPositions) {
+    const _ = this.context.localize;
+
+    const addAction = {
+      label: _('Trace'),
+      handler: ({onUpdate}) => {
+        if (onUpdate) {
+          onUpdate({
+            type: EDITOR_ACTIONS.ADD_TRACE,
+          });
+        }
+      },
+    };
+
+    return (
+      <PlotlyPanel addAction={addAction}>
+        {filteredTraces.length
+          ? this.renderIndividualTraceFolds(
+              filteredTraces,
+              filteredTracesFullDataPositions
+            )
+          : null}
+      </PlotlyPanel>
+    );
+  }
+
+  render() {
+    const {data = [], localize: _, fullData = []} = this.context;
+    const {canAdd, canGroup, excludeFits, useFullData} = this.props;
+    const base = useFullData ? fullData : data;
+
+    // we don't want to include analysis transforms when we're in the create panel
+    const filteredTracesFullDataPositions = [];
+    const filteredTraces = base.filter((t, i) => {
+      if (excludeFits) {
+        return !(t.transforms && t.transforms.every(tr => tr.type === 'fit'));
+      }
+      filteredTracesFullDataPositions.push(i);
+      return true;
+    });
+
+    const groupedTraces = filteredTraces.reduce(
+      (allTraces, nextTrace, index) => {
+        const traceType = plotlyTraceToCustomTrace(nextTrace);
+        const adjustedIndex = useFullData ? nextTrace.index : index;
+        if (!allTraces[traceType]) {
+          allTraces[traceType] = [];
+        }
+        allTraces[traceType].push(adjustedIndex);
+        return allTraces;
+      },
+      {}
+    );
 
-    if (canGroup && filteredData.length > 1 && groupedTraces.length > 0) {
+    if (canAdd) {
```

Comment:
This is basically an early-return case that means we don't need `groupedTraces` so let's move it up above the computation of `groupedTraces` and save a bit of time when `canAdd`